### PR TITLE
Implement tool for reading shares from YAML file 

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR implements `SharesReader` - the tool for reading the `Share` instances from the YAML file. This is needed because of an idea to store shares that will be used in the application, in a file which is more convenient in usage than simple static storage.

### CODE REVIEW

File: model/src/main/java/io/spine/examples/shareaware/sharesreader/SharesReader.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        MapLikeType mapType = mapper.getTypeFactory()
+                                    .constructMapLikeType(Map.class, String.class, String.class);
+        JavaType listType = mapper.getTypeFactory()
+                                  .constructCollectionType(List.class, mapType);
+        try {
+            List<Map<String, String>> sharesData = mapper.readValue(file, listType);
+            return sharesData.stream()
+                    .map(SharesReader::toShare)
+                    .collect(Collectors.toSet());
```
Comment (Vladyslav-Kuksiuk) [inline model/src/main/java/io/spine/examples/shareaware/sharesreader/SharesReader.java:86]: Please import `toSet()` statically.

File: model/src/main/java/io/spine/examples/shareaware/sharesreader/SharesReader.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        } catch (IOException e) {
+            throw illegalStateWithCauseOf(e);
```
Comment (Vladyslav-Kuksiuk) [inline model/src/main/java/io/spine/examples/shareaware/sharesreader/SharesReader.java:88]: I think it should be `IllegalArgumentException`.

**Review by Vladyslav-Kuksiuk — COMMENTED** [review]
Mostly LGTM, with minor comments.

File: buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/Jackson.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.dependency
+
+object Jackson {
```
Comment (armiol) [inline buildSrc/src/main/kotlin/io/spine/examples/shareaware/dependency/Jackson.kt:29]: We usually have a comment which points to the original GitHub repo, or website.

See [this one](https://github.com/SpineEventEngine/config/blob/master/buildSrc/src/main/kotlin/io/spine/internal/dependency/BouncyCastle.kt), as an example.

File: model/src/main/java/io/spine/examples/shareaware/sharesreader/SharesReader.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+     * Prevents instantiation of this class.
+     */
+    private SharesReader() {
+    }
+
+    /**
+     * Returns the set (to prevent duplication) of shares read from the provided YAML file.
+     *
+     * @implNote Shares must be written to the file in this way:
+     * <ul>
```
Comment (armiol) [inline model/src/main/java/io/spine/examples/shareaware/sharesreader/SharesReader.java:64]: Use  the combination of `<pre>` and `{@code}` instead. Here is [an example](https://github.com/SpineEventEngine/core-java/blob/936f7195cd2118bcc1daa63bc66332c7fc9f21e2/client/src/main/java/io/spine/client/CommandRequest.java#L53).

File: model/src/test/resources/sharesCredentials.yml
Comment (armiol) [inline model/src/test/resources/sharesCredentials.yml:1]: This is not "credentials".

**Review by armiol — CHANGES_REQUESTED** [review]
@Artem-Semenov-dev LGTM in general. Please see my comments though.

File: model/src/test/resources/testingShares.yml
Comment (armiol) [inline model/src/test/resources/testingShares.yml:1]: Please rename to `testing-shares.yml`.

File: model/src/main/proto/spine_examples/shareaware/share/share.proto [hunk 5]
```diff
@@ -26,12 +26,12 @@
 
 syntax = "proto3";
 
-package spine_examples.shareaware;
+package spine_examples.shareaware.share;
```
Comment (armiol) [inline model/src/main/proto/spine_examples/shareaware/share/share.proto:29]: I have asked to adjust the Java package only.

Let's leave the Proto message in the same folder and in the same Proto package.

**Review by armiol — CHANGES_REQUESTED** [review]
@Artem-Semenov-dev we are getting closer. Please see my comments.

**Review by armiol — APPROVED** [review]

**Pull request merged**